pkg/bucketeer/model/evaluation_detail.go

```diff
@@ -0,0 +1,62 @@
+package model
+
+type BKTEvaluationDetail[T EvaluationValue] struct {
+	FeatureID      string
+	FeatureVersion int32
+	UserID         string
+	VariationID    string
+	VariationName  string
+	VariationValue T
+	Reason         EvaluationReason
+}
+
+type EvaluationValue interface {
+	int | int64 | float64 | string | bool | interface{}
+}
+
+type EvaluationReason string
+
+const (
+	EvaluationReasonTarget       EvaluationReason = "TARGET"
+	EvaluationReasonRule         EvaluationReason = "RULE"
+	EvaluationReasonDefault      EvaluationReason = "DEFAULT"
+	EvaluationReasonClient       EvaluationReason = "CLIENT"
+	EvaluationReasonOffVariation EvaluationReason = "OFF_VARIATION"
+	EvaluationReasonPrerequisite EvaluationReason = "PREREQUISITE"
+)
+
+func NewEvaluationDetail[T EvaluationValue](
+	featureID, userID, variationID, variationName string,
+	featureVersion int32,
+	reasonType ReasonType,
+	value T,
+) BKTEvaluationDetail[T] {
+	return BKTEvaluationDetail[T]{
+		FeatureID:      featureID,
+		FeatureVersion: featureVersion,
+		UserID:         userID,
+		VariationID:    variationID,
+		VariationName:  variationName,
+		VariationValue: value,
+		Reason:         convertEvaluationReason(reasonType),
+	}
+}
+
+func convertEvaluationReason(reasonType ReasonType) EvaluationReason {
```

Comment:
Changed convertEvaluationReason() to private.
https://github.com/bucketeer-io/go-server-sdk/pull/136#discussion_r1686555456